### PR TITLE
Adds Support for Tax Identifiers

### PR DESCRIPTION
**Pull request by jchen293**
Add Tax Identifier along with the unit test.

### CODE REVIEW

**Comment by nwithan8** [comment]
Merge #75 before merging this.

File: src/main/java/com/easypost/model/Shipment.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
 	String uspsZone;
 	Map<String, Object> options;
 	List<ShipmentMessage> messages;
+    List<TaxIdentifier> taxIdentifiers;
```
Comment (nwithan8) [inline src/main/java/com/easypost/model/Shipment.java:38]: If this looks unindented to you, it's just a GitHub thing.

Will verify with future code styling cleanup.

**Review by nwithan8 — DISMISSED** [review]
Ignoring the #75 -related code, LGTM.

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,7 @@ src/test/test.iml
 
 *.class
 settings.xml
+
+
+src/scratch
+src/scratch/*
```
Comment (jontsai) [inline .gitignore:23]: Always include a newline at the end of the file

Was the previous pattern, `src/scratch` not sufficient to also cover this?

File: src/main/java/com/easypost/model/Shipment.java (hunk 1 above)
Comment (jontsai) [reply src/main/java/com/easypost/model/Shipment.java:38]: The indentation is also off if you view the raw file: https://github.com/EasyPost/easypost-java/blob/59a6fb059e8f4056e2f5b9444fe8728ca58f0e8f/src/main/java/com/easypost/model/Shipment.java

There's a mix of tabs and spaces being used in this file. The existing indentation character in this file appears to use `<tab>`, but I would prefer that we eventually move to using (4?) spaces.

File: src/main/java/com/easypost/model/Tracker.java [hunk 3]
```diff
@@ -148,7 +149,16 @@ public static boolean createList(Map<String, Object> params) throws EasyPostExce
 	public static boolean createList(Map<String, Object> params, String apiKey) throws EasyPostException {
 		String createListUrl = String.format("%s/create_list", classURL(Tracker.class));
 
-		request(RequestMethod.POST, createListUrl, params, Object.class, apiKey);
+		int count = 0;
+		Map<String, Object> newParams = new HashMap<String, Object>();
+		Map<String, Object> trackers = new HashMap<String, Object>();
+		for (Object tracker : (ArrayList)params.get("trackers")) {
```
Comment (jontsai) [inline src/main/java/com/easypost/model/Tracker.java:155]: Should there be a space when type-casting `(ArrayList)params...` -> `(ArrayList) params...` ?

File: src/main/java/com/easypost/model/Tracker.java [hunk 4]
```diff
@@ -148,7 +149,16 @@ public static boolean createList(Map<String, Object> params) throws EasyPostExce
...
+			trackers.put(String.valueOf(count), tracker);
+			count++;
```
Comment (jontsai) [inline src/main/java/com/easypost/model/Tracker.java:157]: Why is it necessary to put `count`, are we not using JSON automatic serialization?

Also, this PR mentions adding tax identifiers, but this change is on `Tracker.java` -- should this change be broken out into a separate PR?

**Review by jontsai — CHANGES_REQUESTED** [review]

File: src/main/java/com/easypost/model/Tracker.java (hunk 4 above)
Comment (nwithan8) [reply src/main/java/com/easypost/model/Tracker.java:157]: Tracker requires a specific format that deviates from the normal format. We need to make a dictionary of number keys and data values. This formats the data properly before calling the HTTP request.

This update is more related to the JSON support PR #75 , the changes of which were accidentally included in this branch. If we merge #75 first as planned, this will be irrelevant.

File: src/main/java/com/easypost/model/Tracker.java (hunk 3 above)
Comment (nwithan8) [reply src/main/java/com/easypost/model/Tracker.java:155]: We can run a linter on the entire library to clean this up in the future.

Comment (nwithan8) [reply src/main/java/com/easypost/model/Tracker.java:155]: New `indent_cleanup` branch for this: https://github.com/EasyPost/easypost-java/tree/indent_cleanup

File: .gitignore (hunk 2 above)
Comment (nwithan8) [reply .gitignore]: Fixed in 927a2c4d7e0953cd65c38ec3e1919aef1b05c616

File: src/main/java/com/easypost/model/Shipment.java (hunk 1 above)
Comment (nwithan8) [reply src/main/java/com/easypost/model/Shipment.java:38]: New indent_cleanup branch for this: https://github.com/EasyPost/easypost-java/tree/indent_cleanup

**Review by Justintime50 — CHANGES_REQUESTED** [review]
I can't reliably take a look at this till it's properly rebased off the JSON encoding changes and no longer includes those in this PR. Blocking till we get there - also threw it in draft mode because it's not ready yet.

File: src/main/java/com/easypost/model/Shipment.java [hunk 5]
```diff
@@ -8,287 +8,295 @@
...
+    String trackingCode;
+    String status;
+    String refundStatus;
+    String batchId;
+    String batchStatus;
+    String batchMessage;
+    String uspsZone;
+    Map<String, Object> options;
+    List<ShipmentMessage> messages;
+    List<TaxIdentifier> taxIdentifiers;
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/Shipment.java:38]: # Reviewer Note

This is the main change, adding `taxIdentifiers` to the shipment object.

File: src/main/java/com/easypost/model/Shipment.java [hunk 6]
```diff
@@ -8,287 +8,295 @@
...
+    public void setForms(List<Form> forms) {
+        this.forms = forms;
+    }
+
+    public List<TaxIdentifier> getTaxIdentifiers(){
+        return taxIdentifiers;
+    }
+    public void setTaxIdentifiers(List<TaxIdentifier> taxIdentifiers){
+        this.taxIdentifiers = taxIdentifiers;
+    }
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/Shipment.java:234]: # Reviewer Note

This is the only other change on this file.

**Review by jontsai — APPROVED** [review]

**Review by Justintime50 — APPROVED** [review]
Unit tests were added, rebased correctly, code looks great!

**Review by nwithan8 — APPROVED** [review]
LGTM!
